packages/manager/modules/pci/src/components/project/instance/instance.class.js

```diff
@@ -196,14 +198,37 @@ export default class Instance {
    * @return {Boolean}
    */
   isDeleting() {
-    return this.status === 'DELETING';
+    return this.status === INSTANCE_STATUS.DELETING;
   }
 
   isStarted() {
-    return this.status === 'ACTIVE';
+    return this.status === INSTANCE_STATUS.ACTIVE;
   }
 
   isStopped() {
-    return this.status === 'SHUTOFF';
+    return this.status === INSTANCE_STATUS.SHUTOFF;
+  }
+
+  isShelved() {
+    return [
+      INSTANCE_STATUS.SHELVED,
+      INSTANCE_STATUS.SHELVED_OFFLOADED,
+    ].includes(this.status);
+  }
+
+  isShelving() {
+    return this.status === INSTANCE_STATUS.SHELVING;
+  }
+
+  isUnshelving() {
+    return this.status === INSTANCE_STATUS.UNSHELVING;
+  }
+
+  isValidToShelve() {
+    return !this.isShelved() && !this.isShelving() && !this.isUnshelving();
```

Comment:
is it possible to shelve an instance that is in deleting, error, or not an active state?

Reply:
I also asked APi team to know that but i don't get response.
As i know when customer shelve an instance he shouldn't unshelve it until shelve is finished and vice versa.

Reply:
So i asked them again, the PCI team said no task can be launched if shelve/unshelve is running.

Reply:
I update both (instances/instance).html pages

Reply:
thanks @radireddy.
Updated.